packages/dds/map/src/directory.ts

```diff
@@ -780,8 +782,8 @@ export class SharedDirectory
 				localOpMetadata,
 			) => {
 				const subdir = this.getWorkingDirectory(op.path) as SubDirectory | undefined;
-				// If there is pending delete op for this subDirectory, then don't apply the this op as we are going
-				// to delete this subDirectory.
+				// If there is pending delete op for any parent of this subdirectory, then don't apply the this op
```

Comment:
```suggestion
				// If there is pending delete op for this subdirectory or any parent, then don't apply the this op
```

It's ambiguous from the comment that the subdirectory itself is included. Same for other comments below.